source/actors/player/Hero.hx

```diff
@@ -75,4 +79,50 @@ class Hero extends FlxSprite {
             velocity.y = 0;
         }
     }
+
+    /**
+	 * Override of FlxSprite hurt function.
+     * @author Matt Lippelman
+     * @param damageAmount an integer value in fragments (4 per heart) to be hurt the player
+     * @return void
+	 */
+    public override function hurt(damageAmount:Float):Void {
+        super.hurt(damageAmount);
+        playerHealth.updateGraphics(health);
+    }
+
+    /**
+	 * Function to heal the player.
+     * @author Matt Lippelman
+     * @param healAmount an integer value in fragments (4 per heart) to be heal the player
+     * @return void
+	 */
+    public function heal(healAmount:Int):Void {
+        super.hurt(-healAmount);
+        if (health >= playerHealth.maxHealth) {
+            health = playerHealth.maxHealth;
+        }
+        playerHealth.updateGraphics(health);
+    }
+
+    /**
+	 * Function to increase maximum health of the player.
+     * @author Matt Lippelman
+     * @param fragments an integer value in fragments (4 per heart) to be added to max health
+     * @return void
+	 */
+    public function increaseMaxHealth(heartFragments:Int):Void {
+        playerHealth.increaseMaxHealth(heartFragments, health);
+        heal(heartFragments);
+    }
+
+    /**
+	 * Function to increase maximum health by one full heart container.
+     * @author Matt Lippelman
+     * @return void
+	 */
+    public function increaseMaxHealthByOneHeart():Void {
+        playerHealth.increaseMaxHealth(HeartContainer.MAX_FRAG_AMOUNT, health);
+        heal(HeartContainer.MAX_FRAG_AMOUNT);
+    }
```

Comment:
I'm curious why we have both - I guess the top one gives us flexibility if we need it, but we don't necessarily have a use case for it yet?